src/Jarvis.Core/Diagnostics/TimedOperationExtensions.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to Spectre Systems AB under one or more agreements.
+// Spectre Systems AB licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Jarvis.Core.Diagnostics
+{
+    public static class TimedOperationExtensions
+    {
+        public static IDisposable TimedOperation(this IJarvisLog logger, string messageTemplate,
```

Comment:
Could we rename this to simply `Timed`?

Reply:
There are a few reasons I sided on `TimedOperation`.

The usage in this PR is only implicit, but the timed operation also support different logging levels depending on if it is completed or cancelled:

```csharp
using (var operation = _logger.TimedOperation("Perform task", LogLevel.Information, LogLevel.Warning))
{
    try
    {
        // perform task, external calls etc.
    }
    catch (Exception e)
    {
        operation.Cancel(); // logged as warning
    }
}
```

I see now that the timed operation only returns an `IDisposable`, so the example above is not applicable now, but I think it wouldn't hurt to expose `Operation` here?

By introducing the concept of "operation", it makes sense that it can be cancelled and completed. It also make sense to enricht the scope/log context with an `operationId`.